lib/package.py

```diff
@@ -144,6 +144,8 @@ def _load(self):
                     source[source_name]['ref_to_fetch'] = None
                 if 'archive_src_dir' not in source.values()[0]:
                     source[source_name]['archive_src_dir'] = None
+                if 'shallow' not in source.values()[0]:
+                    source[source_name]['shallow'] = False
```

Comment:
Perhaps we can have `--git-shallow` (or whatever option name makes sense here) implemented in `host_os.py` script to use git shallow clones.

Or we can implement a `git_args` parameter in `config.yaml` file and add git shallow options there; similar to what we did for `mock_args`.